packages/react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -1023,6 +1054,42 @@ class Dropdown extends AutoControlledComponent<Extendable<DropdownProps>, Dropdo
   private isValueEmpty = (value: ShorthandValue | ShorthandCollection) => {
     return _.isArray(value) ? value.length < 1 : !value
   }
+
+  private getHighlightedIndexOnArrowKeyOpen = (
+    changes: StateChangeOptions<ShorthandValue>,
+  ): number => {
+    // if open by ArrowUp, index should change by -1.
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowUp) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      if (_.isNumber(this.state.highlightedIndex)) {
+        const newIndex = this.state.highlightedIndex - 1
+        return newIndex < 0 ? itemsLength - 1 : newIndex
+      }
+      return itemsLength - 1
+    }
+    // if open by ArrowDown, index should change by +1.
+    if (changes.type === Downshift.stateChangeTypes.keyDownArrowDown) {
+      const itemsLength = this.getItemsFilteredBySearchQuery().length
+      if (_.isNumber(this.state.highlightedIndex)) {
+        const newIndex = this.state.highlightedIndex + 1
+        return newIndex >= itemsLength ? 0 : newIndex
+      }
+      return 0
+    }
+    // if not opened by Arrow, index should not change at all.
+    return undefined
+  }
+
+  private getHighlightedIndexOnClose = (): number => {
```

Comment:
NIT: for readability:
```typescript
private getHighlightedIndexOnClose = (): number => {
  const { highlightFirstItemOnOpen, items, multiple, search } = this.props
  const { value } = this.state

  if (!multiple && !search && value) {
    // in single selection, if there is a selected item, highlight it.
    return items.indexOf(value)
  }

  if (highlightFirstItemOnOpen) {
    // otherwise, if highlightFirstItemOnOpen prop is provied, highlight first item
    return 0
  }

  // otherwise, highlight no item
  return null
}
```

Reply:
thanks, will use that version